### PR TITLE
[Improve][Connector-v2] Redis support select db

### PR DESCRIPTION
**Pull request by xiaofan2022**
### Purpose of this pull request
redis select_db index #5415
### Does this PR introduce _any_ user-facing change?
no
### How was this patch tested?
add new test

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/connector-v2/sink/Redis.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ Used to write data to Redis.
 | data_type      | string | yes                   | -             |
 | user           | string | no                    | -             |
 | auth           | string | no                    | -             |
+| db_num         | int    | no                    | -             |
```
Comment (TyrantLucifer) [inline docs/en/connector-v2/sink/Redis.md]:
```suggestion
| db_num         | int    | no                    | 0            |
```

File: seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java [hunk 2]
```diff
@@ -40,6 +40,7 @@ public class RedisParameters implements Serializable {
     private String host;
     private int port;
     private String auth = "";
+    private Integer dbNum;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java]:
```suggestion
    private int dbNum;
```

File: seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java [hunk 3]
```diff
@@ -115,6 +120,9 @@ public Jedis buildJedis() {
                 if (StringUtils.isNotBlank(user)) {
                     jedis.aclSetUser(user);
                 }
+                if (dbNum != null && dbNum > 0) {
+                    jedis.select(dbNum);
+                }
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java]:
```suggestion
                    jedis.select(dbNum);
```

File: seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java [hunk 4]
```diff
@@ -148,7 +156,11 @@ public Jedis buildJedis() {
                 } else {
                     jedisCluster = new JedisCluster(nodes);
                 }
-                return new JedisWrapper(jedisCluster);
+                JedisWrapper jedisWrapper = new JedisWrapper(jedisCluster);
+                if (dbNum != null && dbNum > 0) {
+                    jedisWrapper.select(dbNum);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java]: The same as above.

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
Thank you for your contribution, overall LGTM, but you need offer e2e test cases for this change to verify it worked.

File: seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java (hunk 3 above)
Comment (xiaofan2022) [reply seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java]: done

File: seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java (hunk 2 above)
Comment (xiaofan2022) [reply seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java]: done

**Review by xiaofan2022 — COMMENTED** [review]
done

**Comment by hailin0** [comment]
update PR title ?

**Review by xiaofan2022 — COMMENTED** [review]
done

**Review by xiaofan2022 — COMMENTED** [review]
done

File: seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisConfig.java [hunk 5]
```diff
@@ -48,6 +48,13 @@ public enum HashKeyParseMode {
                     .withDescription(
                             "redis authentication password, you need it when you connect to an encrypted cluster");
 
+    public static final Option<Integer> DB_NUM =
+            Options.key("db_num")
+                    .intType()
+                    .defaultValue(0)
+                    .withDescription(
+                            "redis  database index id, It is connected to db 0 by default");
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisConfig.java]:
```suggestion
                            "Redis database index id, it is connected to db 0 by default");
```

Comment (xiaofan2022) [reply seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisConfig.java]: done

**Review by Hisoka-X — APPROVED** [review]
LGTM, but I think @TyrantLucifer should take a look.

**Comment by xiaofan2022** [comment]
> Thank you for your contribution, overall LGTM, but you need offer e2e test cases for this change to verify it worked.

done

**Review by hailin0 — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Comment by Hisoka-X** [comment]
Thanks @xiaofan2022 and @hailin0 @TyrantLucifer for review.

**Pull request merged**